src/recursion.js

```diff
@@ -3,17 +3,130 @@
 const nFibonacci = (n) => {
   // fibonacci sequence: 1 2 3 5 8 13 ...
   // return the nth number in the sequence
+  if (n < 0) {
+    return 0;
+  } else if (n === 0) {
+    return 0;
+  } else if (n === 1) {
+    return 1;
+  } else if (n === 2) {
+    return 2;
+  }
+  return nFibonacci(n - 1) + nFibonacci(n - 2);
 };
 
 const nFactorial = (n) => {
   // factorial example: !5 = 5 * 4 * 3 * 2 * 1
   // return the factorial of `n`
+  if (n > 0) {
+    return n * nFactorial(n - 1);
+  }
+  return 1;
 };
 
 /* Extra Credit */
+
 const checkMatchingLeaves = (obj) => {
   // return true if every property on `obj` is the same
   // otherwise return false
+  let level = 0;
+  let counter = 0;
+
+  function hasNoObject(someObject) {
+    Object.values(someObject).forEach((val) => {
+      if (typeof val === 'object') {
+        // console.log(`Check: Object found in values. (@ depth level:${level})`);
+        return false;
+      }
+    });
+    // console.log(`Check: No object found in values. (@ depth level:${level})`);
+    return true;
+    // ~~~~~~~~WHYYY!!!!!! Does this happen!!!!!!~~~~~
```

Comment:
You're going to be receiving this as a toy problem today. You should be able to think through it. Otherwise you'll have some exposure to it and you'll be able to come back and nail it!